packages/discord.js/src/structures/AutoModerationRule.js

```diff
@@ -190,6 +193,17 @@ class AutoModerationRule extends Base {
     return this.edit({ triggerMetadata: { keywordFilter }, reason });
   }
 
+  /**
+   * Sets the regular expression patterns for this auto moderation rule.
+   * @param {string[]} regexPatterns The regular expression patterns of this auto moderation rule
+   * <info>Only Rust-flavored regular expressions are supported.</info>
+   * @param {string} [reason] The reason for changing the regular expression patterns of this auto moderation rule
+   * @returns {Promise<AutoModerationRule>}
+   */
+  setRegexPatterns(regexPatterns, reason) {
```

Comment:
We should allow regex expressions as well and transform them into strings. This would enable the end user to have regex syntax highlighting in their code editor.

Reply:
I disagree. Rust-flavoured regular expressions are not the same as the ones used in JavaScript. This will lead to linting errors, if linting is used, as well as a false perception of syntax highlighting.

Reply:
I ran some tests myself and there are some similarities between Rust's regular expressions and JavaScript's, but they are not identical. A warning in the JS documentation that it will not be run by a JavaScript regex engine might be helpful. (Alternatively, the user should use another library that converts JS Regex expressions to Rust-flavored expressions anyways).

Reply:
I'm with Jira here, we should absolutely not be taking in `RegExp` unless its 1-to-1 with what Discord expects. It's straight up miss-leading - even with a doc warning.

Reply:
There are quite a few differences between Rust and JavaScript regular expressions, I agree 100% that `RegExp`s shouldn't be allowed as it will just cause confusion